feat: Integrate Bitnami MariaDB chart with conditional deployment

Awungia112 opened the pull request:
Adds MariaDB chart dependency to enable in-cluster MariaDB database deployment for Moodle.

## Changes
- Add MariaDB dependency (v21.0.7) with conditional deployment
- Update values.yaml with enhanced configuration options
- Improve documentation and usage examples
- Maintain backward compatibility

## Testing
✅ Helm dependency update  
✅ Template validation  
✅ Local k3s deployment  
✅ MariaDB connectivity

Awungia112 committed:
chore: update Chart.lock with new generated timestamp

Awungia112 committed:
chore: remove values-cnpg.yaml file for CloudNativePG configuration

Awungia112 committed:
feat: add MariaDB configuration for Moodle Helm chart

- Introduced a new values file `values-mariadb.yaml` for configuring MariaDB settings.
- Included authentication details such as root password, database name, username, and password.
- Enabled persistence for the primary database instance with a specified storage class.

Awungia112 committed:
chore: disable PostgreSQL support in Moodle Helm chart configuration

- Set the `enabled` flag for PostgreSQL to false in `values-postgres.yaml` to disable its usage in the Helm chart.

Awungia112 committed:
chore: enable MariaDB support and disable CloudNativePG in Moodle Helm chart configuration

- Set the `enabled` flag for MariaDB to true in `values.yaml` to activate its usage.
- Disabled CloudNativePG support by setting its `enabled` flag to false.

marcjazz commented on charts/moodle/values-mariadb.yaml:
```diff
@@ -0,0 +1,13 @@
+   
+    username: bn_moodle
+   
+    password: moodlepass
```
What is the plan with the space?

marcjazz reviewed (changes requested):
Good job just have few remarks: 
- Postgres is started when disabled
- Also notice that the moodle is installed within the default namespace
- Postgres doesn't work with the values-portgres.yaml that is provided. 
- Hmmmm, nothing seems to be working as expected


<img width="1837" height="99" alt="image" src="https://github.com/user-attachments/assets/3d4720c0-2662-4d6f-b16e-7940fed06e07" />

Awungia112 committed:
chore: update Chart.lock to remove CloudNativePG dependency and regenerate digest

- Removed CloudNativePG dependency from Chart.lock.
- Updated digest and generated timestamp for the lock file.

Awungia112 committed:
chore: update Chart.yaml to conditionally enable PostgreSQL dependency

- Modified the PostgreSQL dependency entry to include a condition for enabling it.
- Removed the direct listing of the CloudNativePG dependency as it is no longer needed.

Awungia112 committed:
chore: update README.md to reflect changes in database support for Moodle Helm chart

- Updated documentation to clarify support for both PostgreSQL and MariaDB as internal database options.
- Removed references to CloudNativePG and adjusted installation commands to include MariaDB configuration.
- Enhanced instructions for enabling the desired database in the values files.

Awungia112 committed:
chore: remove PostgreSQL and CloudNativePG configurations from Moodle Helm chart values

- Disabled PostgreSQL support by removing its configuration from `values.yaml`.
- Eliminated CloudNativePG references to streamline database options for Moodle, focusing on MariaDB.

Awungia112 committed:
chore: update README.md installation instructions for Moodle Helm chart

- Enhanced the installation commands to include namespace creation and improved formatting for clarity.
- Updated the dry run command to reflect the namespace option for consistency with the installation instructions.

Awungia112 committed:
chore: clean up MariaDB configuration in values-mariadb.yaml

- Removed unnecessary whitespace and redundant lines in the MariaDB configuration section.
- Streamlined the authentication details for clarity.

stephane-segning commented on charts/moodle/Chart.yaml:
Why did you removed `cloudnative-pg` ?

Awungia112 replied:
Hi @stephane-segning I removed the cloudnative-pg temporarily because it hasn't been properly implemented yet and i plan to re-introduce it when i am done with that ticket

stephane-segning replied:
Ok, then please create a ticket for that, or re-open the corresponding ticket.

stephane-segning commented on charts/moodle/README.md:
````diff
@@ -38,39 +37,28 @@ helm dependency build
+#### NB
+The database deployed will be based on one you enable as true.
 
+``` postgresql/mariadb: 
````
Wrong format here

stephane-segning commented on charts/moodle/README.md:
````diff
@@ -86,7 +74,7 @@ helm uninstall my-moodle
 
 ```bash
-helm template my-moodle . --values values.yaml
+helm template  my-moodle . -n moodle --values values.yaml --values values-mariadb.yaml --values values-postgres.yaml
````
Why mariab and postgres values at the same time?

Awungia112 replied:
Thank you @stephane-segning  
I used both the mariadb and postgres because the helm template output comes entirely from which values files you pass, because Helm merges them in order and uses those to decide which Kubernetes objects to render. Also the postgres.enabled condition  which i set as false is found in the vlaues-postgres.yaml

stephane-segning commented on charts/moodle/README.md:
```diff
@@ -159,7 +146,7 @@ helm dependency update
 * `values.yaml` – for use with external databases (default)
 * `values-postgres.yaml` – enables internal PostgreSQL (Bitnami)
-* `values-postgres.yaml`– used to enable CloudNativePG(Bitnami)  
+* `values-mariadb.yaml`– used to pass mariadb autherntication values  
```
No, used to enable and configure mariadb

stephane-segning commented on charts/moodle/values-cnpg.yaml:
Why did you removed cnpg?

Awungia112 replied:
Hello @stephane-segning  you said we won't be using the cnpg helm chart that is why i removed it

stephane-segning replied:
We'll be using the values, not the helm chart. And I think it's not the good ticket to work on here

stephane-segning commented on charts/moodle/README.md:
````diff
@@ -38,39 +37,28 @@ helm dependency build
 ```
 
-### Option 2: Internal PostgreSQL (Bitnami dependency)
+### Option 2: Internal PostgreSQL Or Mariadb (Bitnami dependency)
````
No, postgres is along mariadb. But never the two at the same time.

stephane-segning commented on charts/moodle/README.md:
Why do you remove cnpg ?

stephane-segning commented on charts/moodle/values-postgres.yaml:
Wrong. Postgres should be activated by its own value- file

stephane-segning commented on charts/moodle/values.yaml:
```diff
@@ -7,10 +7,7 @@ moodle:
   mariadb:
-    enabled: false
-
-  postgresql:
```
Why disabling this? Keep it. But disable it.

Awungia112 replied:
Hello @stephane-segning this line is not being used because the chart.yaml instructs the chart to look for the root key postgres (- condition: postgresql.enabled)  before checking the value of the (true or false) condition.  
The root key "postgres" is only found in the values-postgres.yaml. 
The root key in the values.yaml is "moodle".

stephane-segning replied:
Yeah but the main values.yaml should define defaults for all others.

stephane-segning replied:
So they should be defined, but not enabled

stephane-segning commented on charts/moodle/values.yaml:
```diff
@@ -22,6 +19,3 @@ moodle:
     password: moodlepass
     database: moodledb
 
-cloudnative-pg:
```
Why disabling this? Keep it. But disable it.

Awungia112 committed:
chore: update README.md to clarify database options and installation instructions for Moodle Helm chart

Awungia112 committed:
chore: update MariaDB username and enable persistence in values-mariadb.yaml

Awungia112 committed:
chore: update installation instructions in README.md for MariaDB and PostgreSQL

Awungia112 committed:
chore: disable persistence for MariaDB in values-mariadb.yaml

Awungia112 committed:
chore: update PostgreSQL configuration in values-postgres.yaml for Moodle

Awungia112 committed:
chore: update MariaDB configuration in values.yaml for Moodle

stephane-segning commented on charts/moodle/values-postgres.yaml:
```diff
@@ -1,19 +1,9 @@
-postgresql:
-  enabled: true
```
Why did you removed the key `enabled: true` ?

Awungia112 committed:
chore: ensure MariaDB is enabled in values-mariadb.yaml

Awungia112 committed:
chore: update PostgreSQL enabled status in values-postgres.yaml

Awungia112 committed:
chore: disable MariaDB and update service configuration in values.yaml

Awungia112 committed:
chore: enable PostgreSQL in values-postgres.yaml

stephane-segning commented on charts/moodle/values.yaml:
```diff
@@ -5,23 +5,26 @@ moodle:
+    port: 5432
+    user: bn_moodle_adorsys
     password: moodlepass
     database: moodledb
```
Put this block inside of the corresponding db value- file

stephane-segning commented on charts/moodle/values.yaml:
I can notice `postgresql:` two times in this file?

Awungia112 committed:
chore: remove external database configuration for PostgreSQL in values.yaml

Awungia112 committed:
chore: reorganize external database configuration in values-postgres.yaml

Awungia112 committed:
chore: clean up formatting and whitespace in values.yaml

stephane-segning commented on charts/moodle/values.yaml:
Won't you also define a key for `cnpg` ?

Awungia112 replied:
I won’t add a cnpg key, since we aren’t using the CNPG Helm chart here  a root key like cnpg would have no effect unless the templates reference it. I’ll treat CNPG as an external database and configure it via the existing moodle.externalDatabase settings.

stephane-segning replied:
You don't need to have the CNPG Helm chart to add it here like that. You can add it here and use it as I shown you yesterday, to enable a custom manifest, in this case, a custom CNPG Cluster Manifest.

Awungia112 committed:
chore: add cnpg configuration and maintain PostgreSQL disabled status in values.yaml

stephane-segning commented:
@codex review this pr point by point

marcjazz commented:
Could you please resolve the merge conflicts first

Awungia112 committed:
Merge branch 'main' into feature/mariadb-integration

Awungia112 committed:
docs: Update README for Moodle Helm Chart to clarify service type configuration and add dependencies section

Awungia112 committed:
chore: update Chart.lock with new generated timestamp

Awungia112 committed:
refactor: clean up values.yaml by removing redundant PostgreSQL configuration and comments

Awungia112 committed:
chore: remove cloudnative-pg dependency from Chart.yaml to streamline Moodle Helm chart configuration

Awungia112 committed:
refactor: update values.yaml to disable PostgreSQL and clean up database configuration

stephane-segning commented on charts/moodle/values.yaml:
```diff
@@ -3,26 +3,17 @@ moodle:
-
   persistence:
-    enabled: true
+    enabled: false
```
Are you sure to disable this?

Awungia112 replied:
No,  I only disabled it for testing i will correct it.

stephane-segning commented:
@codex check this pr

Awungia112 commented on charts/moodle/values-postgres.yaml:
```diff
@@ -0,0 +1,21 @@
+moodle:
+   externalDatabase:
+    host: moodle-postgresql-rw
+    user: moodle
+    type: pgsql
+    host: my-moodle-postgresql
+    port: 5432
+    user: bn_moodle_adorsys
+    password: moodlepass
+    database: moodledb
```
- The ```external database``` block if set to true will tell moodle to use a  database outside the chart whereas the ```pgsql``` is  part of the chart hence i didn't see the need for a ```moodle.externalDatabase.enabled``` . 

- The  ```moodle.moodleDatabaseType``` is not necessary because a  ```moodle.externaldatabase.type```  already exists.
 
- The issue of moodle defaulting to  MariaDB will not arise because ```MariaDB``` has already been enabled as false in the default values.yaml file and in the event where moodle is unable to read the ```pgsql``` it won't default to ```MariaDB``` or any database at all, but this is not the case here because for ```pgsql```  moodle will be deployed using   ``` helm install my-moodle . \
  -n moodle \
  --create-namespace \
  -f values.yaml \
  -f values-postgres.yaml  ``` hence the database pgsql aswell
<img width="1529" height="203" alt="Screenshot from 2025-09-18 14-50-12" src="https://github.com/user-attachments/assets/2784f786-914f-4cd3-99a5-69ddb3505619" />

Awungia112 committed:
refactor: update values.yaml to enable persistence

marcjazz commented on charts/moodle/values.yaml:
```diff
@@ -3,26 +3,17 @@ moodle:
 postgresql:
-  enabled: false
-
+    enabled: false    
```
hmm, the indentation is not good here.

marcjazz reviewed (commented):
Looks good, if not the code formatting.

Awungia112 committed:
Fix: removed indentation error
